apps/extension/src/core/managers/config.ts

```diff
@@ -1,98 +1,256 @@
       Extension.sendToBackground(PortName.Events, {
         request: {
           event: EventType.ModelChanged,
-          data: { model: id }
+          data: { model: configManager.getCurrentModel(config) }
         }
       })
     }
   }
 
   async getDefault(): Promise<Config> {
-    let id = (await this.defaultConfig.get("id")) as ModelID | undefined
-    if (!id) {
-      id = ModelID.GPT3
-      await this.setDefault(id)
+    const id = (await this.defaultConfig.get("id")) as string | undefined
+    if (id) {
+      const config = await this.get(id)
+      if (config) {
+        return config
+      }
+      await this.defaultConfig.remove("id")
     }
-    return (await this.get(id)) || this.init(id)
+    return this.init(AuthType.External)
   }
 
   // TODO: allow multiple custom models
-  async getWithDefault(model?: string): Promise<Config> {
+  async forModelWithDefault(model?: string): Promise<Config> {
     if (!model) {
       return this.getDefault()
     }
-    if ((Object.values(ModelID) as string[]).includes(model)) {
-      return this.getOrInit(model as ModelID)
+    if (isKnownModel(model)) {
+      return this.forModel(model)
+    }
+    // TEMP: Handle unknown models using one custom model
+    const configs = await this.filter({
+      auth: AuthType.APIKey,
+      model: null
+    })
+    if (configs.length > 0) {
+      return configs[0]
+    }
+    return this.init(AuthType.APIKey)
+  }
+
+  // Filtering for `null` looks for configs that don't have any models
+  async filter({
+    auth,
+    model
+  }: {
+    auth: AuthType
+    model?: ModelID | null
+  }): Promise<Config[]> {
+    const ids = await this.getIds(100, 0, authIndexName, auth)
+    const maybeConfigs = await Promise.all(ids.map((id) => this.get(id)))
+    const configs = maybeConfigs.filter((c) => c !== undefined) as Config[]
+    return configs.filter((c) =>
+      model === null
+        ? c.models.length === 0
+        : model
+        ? c.models.includes(model)
+        : true
+    )
+  }
+
+  async forAuthAndModel(auth: AuthType, modelId?: ModelID) {
+    const forAuth = await this.filter({ auth })
+    if (!modelId) {
+      return forAuth[0]
+    }
+    return forAuth.find((c) => c.models.includes(modelId))
+  }
+
+  getCallerForAuth(auth: AuthType, modelId?: ModelID) {
+    switch (auth) {
+      case AuthType.External:
+        return openrouter
+      case AuthType.APIKey:
+        return modelId ? modelAPICallers[modelId] : local
+    }
+  }
+
+  getLabelForAuth(auth: AuthType, modelId?: ModelID) {
+    switch (auth) {
+      case AuthType.External:
+        return "OpenRouter"
+      case AuthType.APIKey:
+        return modelId ? defaultAPILabel[modelId] : "Local"
+    }
+  }
+
+  getCaller(config: Config) {
+    return this.getCallerForAuth(config.auth, this.getCurrentModel(config))
+  }
+
+  getCurrentModel(config: Config): ModelID | undefined {
+    // TODO: support multiple models per config
+    if (config.models.length > 1) {
```

Comment:
Seem like we can just return the first model in the config for now if it has more than 1?

Reply:
I thought "undefined" was more appropriate because the external model might have a different one configured, and without making an API call to it (can add that later perhaps), we don't know which one is currently set.